openshift-ci: Add script to build and deploy the webhook

wainersm opened the pull request:
The deploy_webhook.sh script builds and deploy the kata-webhook in
the OpenShift test cluster.

Fixes #2527
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

wainersm commented:
/test-debian

GabyCT commented on .ci/openshift-ci/cluster/deploy_webhook.sh:
```diff
@@ -0,0 +1,42 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This script builds the kata-webhook and deploy it in the test cluster.
```
/deploy/deploys

GabyCT commented on .ci/openshift-ci/cluster/deploy_webhook.sh:
```diff
@@ -0,0 +1,42 @@
+#
+info "Build the kata-webhook"
+./create-certs.sh
+info "Deploy the kata-webhook"
```
/deploy/deploys

GabyCT commented on .ci/openshift-ci/cluster/deploy_webhook.sh:
```diff
@@ -0,0 +1,42 @@
+pushd "${webhook_dir}" >/dev/null
+# Build and deploy the webhook
+#
+info "Build the kata-webhook"
```
/build/builds

wainersm commented:
Updated the PR to address @GabyCT 's comments. Thanks!

wainersm commented:
/test-debian

snir911 commented on .ci/openshift-ci/cluster/deploy_webhook.sh:
```diff
@@ -0,0 +1,42 @@
+
+# Check the web-hook is working as expected.
+#
+[ oc get pod/${hello_pod_name} &>/dev/null ] && \
```
i didn't get this one, does the pod might exist already?
also i'd check the success of the previous command
fine with me other than that, thanks!

wainersm replied:
> i didn't get this one, does the pod might exist already?

It's unlikely, but you never know :)

> also i'd check the success of the previous command (making sure the waiting didn't timeout)

On failure of previous command the script will exit because it is `set -e`. However I will add a check and gentle fail message.

snir911 reviewed (approved):
LGTM, thanks (I added do-not-merge in case you want to add this check, fine with me either way )

wainersm commented:
> LGTM, thanks (I added do-not-merge in case you want to add this check, fine with me either way )

Great, thanks @snir911 ! Just sent an updated version where I added the check. Once the CI jobs finish, I will merge this.

wainersm commented:
/test-debian

wainersm commented:
/test-ubuntu

wainersm commented:
The error on failed CI job (jenkins-ci-ubuntu-18-04-initrd) has nothing to do with this PR.
